src/org/zaproxy/zap/extension/pscanrulesAlpha/CSPNonceScanner.java

```diff
@@ -0,0 +1,201 @@
+
+	private enum VulnType {CSP_NONCE_REUSED, CSP_INVALID_NONCE};
+
+	private PassiveScanThread parent = null;
+	private static final Logger logger = Logger.getLogger(CSPNonceScanner.class);
+
+	// nonce-srcs seen in responses
+	private HashSet<String> Nonces = new HashSet<String>();
+
+	@Override
+	public void setParent(PassiveScanThread parent) {
+		this.parent = parent;
+	}
+
+	@Override
+	public void scanHttpRequestSend(HttpMessage msg, int id) {
+		// Only checking the response for this plugin
+	}
+
+	private void raiseAlert(VulnType currentVT, String evidence, HttpMessage msg, int id) {
+		Alert alert = new Alert(getPluginId(), //PluginID
+					getRisk(currentVT),
+					Alert.CONFIDENCE_HIGH, //Reliability
+					getAlertElement(currentVT, "name")); //Name
+			alert.setDetail(
+				getAlertElement(currentVT, "desc"), //Description
+				msg.getRequestHeader().getURI().toString(), //URI
+				"",	// Param
+				"", // Attack
+				"", // Other info
+				getAlertElement(currentVT, "soln"), //Solution
+				getAlertElement(currentVT, "refs"), //References
+				evidence,	// Evidence
+					16, // CWE-16: Configuration
+					15,	//WASC-15: Application Misconfiguration
+				msg); //HttpMessage
+			parent.raiseAlert(id, alert);
+	}
+
+	private boolean isNonceSrcValid(String nonceSrc) {
+		Matcher validNonceMatcher = VALID_NONCE_PATT.matcher(nonceSrc);
+		return validNonceMatcher.matches();
+	}
+
+	private List<String> getNonceSrcs(String header) {
+		List<String> nonceSrcs = new ArrayList<String>();
+		Matcher nonceSrcMatcher = NONCE_SRC_PATT.matcher(header);
+
+		while (nonceSrcMatcher.find()) {
+			nonceSrcs.add(nonceSrcMatcher.group(1));
+		}
+		return nonceSrcs;
+	}
+
+	private void checkNonces(HttpMessage msg, int id, String header) {
+		for (String nonceSrc: getNonceSrcs(header)) {
+			if (!isNonceSrcValid(nonceSrc)) {
+				raiseAlert(VulnType.CSP_INVALID_NONCE, nonceSrc, msg, id);
+			}
+			if (Nonces.contains(nonceSrc)) {
+				raiseAlert(VulnType.CSP_NONCE_REUSED, nonceSrc, msg, id);
+			}
+			Nonces.add(nonceSrc);
+		}
+	}
+
+	@Override
+	public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+		long start = System.currentTimeMillis();
+		Vector<String> cspHeaders = msg.getResponseHeader().getHeaders(CSP_HEADER);
+		List<String> metaCSPHeaders = getMetaCSPContent(source);
+
+		// Check CSP HTTP Response Headers
+		if (cspHeaders != null) {
+			for (String header: cspHeaders) {
+				checkNonces(msg, id, header);
+			}
+		}
+
+		// Check meta http-equiv CSP content
+		for (String header: metaCSPHeaders) {
+			checkNonces(msg, id, header);
+		}
+
+		if (logger.isDebugEnabled()) {
+			logger.debug("\tScan of record " + id + " took " + (System.currentTimeMillis() - start) + " ms");
+		}
+	}
+
+	@Override
+	public int getPluginId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "scanner.name");

```

Comment:
You key doesn't contain "scanner"......should use `name` so that the concatenation will match:
`pscanalpha.cspnonce.name = Content Security Policy (CSP) Nonce Scanner`